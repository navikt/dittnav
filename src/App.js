import React, { useEffect } from 'react';
import { Switch, Route, BrowserRouter as Router } from 'react-router-dom';
import { useQuery } from 'react-query';
import { Toggle } from './constants';
import useApiQueries from './hooks/useApiQueries';
import ScrollToTop from './components/scroll/ScrollToTop';
import Home from './pages/Home/Home';
import Varslinger from './pages/Varslinger/Varslinger';
import HendelserTestSide from './pages/Hendelser/HendelserTestSide';
import Statusoppdatering from './pages/Statusoppdatering/Statusoppdatering';
import Tidslinje from './pages/Tidslinje/Tidslinje';
import { fetchUnleashToggle } from './Api';

const App = () => {
  const { data: minSideToggle } = useQuery('minside', fetchUnleashToggle);
  useApiQueries();

  useEffect(() => {
    if (minSideToggle?.content === true) {
      window.location.assign(`${window.env.NAVNO_URL}/minside`);
    }
  }, [minSideToggle]);

  return (
    <Router basename="/person">
      <ScrollToTop />
      <Switch>
        <Route
          path="/dittnav"
          exact
          component={Home}
        />
        <Route
          path="/dittnav/varslinger"
          exact
          component={Varslinger}
        />
        {Toggle.TEST_SIDE && (
          <Route
            path="/dittnav/hendelser"
            exact
            component={HendelserTestSide}
          />
        )}
        {Toggle.TEST_SIDE && (
          <Route
            path="/dittnav/statusoppdatering"
            exact
            component={Statusoppdatering}
          />
        )}
        {Toggle.TEST_SIDE && (
          <Route
            path="/dittnav/tidslinje"
            exact
            component={Tidslinje}
          />
        )}
        <Route
          path="/dittnav*"
          exact
          component={Home}
        />
      </Switch>
    </Router>
  );
};

export default App;
