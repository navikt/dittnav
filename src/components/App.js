import React, { Component } from 'react';
import PersonInfo from './PersonInfo';
import InfoMeldinger from './InfoMeldinger';
import Tjenester from './Tjenester';
import Lenkelister from './Lenkelister';
import Artikler from './Artikler';

// // dittnav\web\src\main\java\no\nav\sbl\dialogarena\dittnav\web\pages\ForsideInnlogget.html
class App extends Component {
  render() {
    return (
      <main role="main">
        <section className="service-notification" style={{ display: 'block' }}>
          <h1 className="typo-element">
            <span style={{ display: 'inline' }}>Det skjedde en feil under henting av saker</span>
            <span style={{ display: 'inline' }}>Det skjedde en feil under henting av meldinger fra din innboks</span>
          </h1>
        </section>
        <div className="container">
          <div className="row">
            <div className="maincontent side-innhold">
              <div className="col-md-12">
                <PersonInfo />
                <InfoMeldinger />
                <Tjenester />
                <Lenkelister />
              </div>
            </div>
          </div>

          <Artikler />

        </div>
      </main>
    );
  }
}

export default App;
