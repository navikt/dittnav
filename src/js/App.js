import React, { useEffect } from 'react';
import { Switch, Route, BrowserRouter as Router } from 'react-router-dom';
import HendelserTestSide from './pages/Hendelser/HendelserTestSide';
import ScrollToTop from './components/scroll/ScrollToTop';
import Config from './globalConfig';
import '../less/index.less';
import Api from './Api';
import useStore from './hooks/useStore';
import Home from './pages/Home/Home';
import Varslinger from './pages/Varslinger/Varslinger';
import checkIfModalShouldBeToggled from './utils/modal';

const App = () => {
  const store = useStore();

  useEffect(
    () => {
      Api.fetchBeskjeder()
        .then(([content, headers]) => {
          checkIfModalShouldBeToggled(headers, store.toggleInnloggingsModal);
          store.addBeskjeder(content);
        })
        .catch(() => {
          store.setErrorBeskjeder();
        });

      Api.fetchOppgaver()
        .then(([content, headers]) => {
          checkIfModalShouldBeToggled(headers, store.toggleInnloggingsModal);
          store.addOppgaver(content);
        })
        .catch(() => {
          store.setErrorOppgaver();
        });

      Api.fetchInnbokser()
        .then(([content, headers]) => {
          checkIfModalShouldBeToggled(headers, store.toggleInnloggingsModal);
          store.addInnbokser(content);
        })
        .catch(() => {
          store.setErrorInnbokser();
        });

      Api.fetchInaktiveBeskjeder()
        .then(([content, headers]) => {
          checkIfModalShouldBeToggled(headers, store.toggleInnloggingsModal);
          store.addInaktiveBeskjeder(content);
        })
        .catch(() => {
          store.setErrorInaktiveBeskjeder();
        });

      Api.fetchInaktiveOppgaver()
        .then(([content, headers]) => {
          checkIfModalShouldBeToggled(headers, store.toggleInnloggingsModal);
          store.addInaktiveOppgaver(content);
        })
        .catch(() => {
          store.setErrorInaktiveOppgaver();
        });

      Api.fetchInaktiveInnbokser()
        .then(([content, headers]) => {
          checkIfModalShouldBeToggled(headers, store.toggleInnloggingsModal);
          store.addInaktiveInnbokser(content);
        })
        .catch(() => {
          store.setErrorInaktiveInnbokser();
        });

      Api.fetchInnloggingsstatus()
        .then(([content]) => {
          store.addInnloggingsstatus(content);
        })
        .catch(() => {
          store.setErrorInnloggingsstatus();
        });
    }, [],
  );

  useEffect(
    () => {
      Api.fetchOppfolging()
        .then(([content, headers]) => {
          checkIfModalShouldBeToggled(headers, store.toggleInnloggingsModal);
          store.addOppfolging(content);
        }).catch(() => {
          store.setErrorOppfolging();
        });

      Api.fetchMeldekort()
        .then(([content, headers]) => {
          checkIfModalShouldBeToggled(headers, store.toggleInnloggingsModal);
          store.addMeldekort(content);
        }).catch(() => {
          store.setErrorMeldekort();
        });

      Api.fetchPersonNavn()
        .then(([content, headers]) => {
          checkIfModalShouldBeToggled(headers, store.toggleInnloggingsModal);
          store.addNavn(content);
        }).catch(() => {
          Api.fetchPersonIdent()
            .then(([content]) => {
              store.addIdent(content);
              store.setErrorNavn();
            })
            .catch(() => {
              store.setErrorIdent();
            });
        });

      Api.fetchSaker()
        .then(([content, headers]) => {
          const { feilendeBaksystem } = content;
          if (feilendeBaksystem.length > 0) {
            store.addPaabegynteSoknader(content);
            store.setErrorPaabegynteSoknader();
          } else {
            checkIfModalShouldBeToggled(headers, store.toggleInnloggingsModal);
            store.addPaabegynteSoknader(content);
          }
        }).catch(() => {
          store.setErrorPaabegynteSoknader();
        });

      Api.fetchMeldinger()
        .then(([content, headers]) => {
          checkIfModalShouldBeToggled(headers, store.toggleInnloggingsModal);
          store.addMeldinger(content);
        }).catch(() => {
          store.setErrorMeldinger();
        });

      Api.fetchSakstema()
        .then(([content, headers]) => {
          checkIfModalShouldBeToggled(headers, store.toggleInnloggingsModal);
          store.addSakstema(content);
        }).catch(() => {
          store.setErrorSakstema();
        });
    }, [],
  );

  return (
    <Router basename="/person">
      <ScrollToTop />
      <Switch>
        <Route
          path="/dittnav"
          exact
          component={Home}
        />
        <Route
          path="/dittnav/varslinger"
          exact
          component={Varslinger}
        />
        {Config.TEST_SIDE_FEATURE_TOGGLE && (
          <Route
            path="/dittnav/hendelser"
            exact
            component={HendelserTestSide}
          />
        )}
        <Route
          path="/dittnav*"
          exact
          component={Home}
        />
      </Switch>
    </Router>
  );
};

export default App;
