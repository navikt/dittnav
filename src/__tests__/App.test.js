import React from 'react';
import ReactDOM from 'react-dom';
import App from 'js/App';

const ReactTestRenderer = require('react-test-renderer');
import ShallowRenderer from 'react-test-renderer/shallow';
import wrapIntl from 'js/IntlTestHelper';

const mockApi = () => {
  return {
    fetchOppfolging: () => new Promise((resolve, reject) => {}),
    fetchMeldekort: () => new Promise((resolve, reject) => {}),
    fetchPersonNavn: () => new Promise((resolve, reject) => {}),
    fetchPersonIdent: () => new Promise((resolve, reject) => {}),
    fetchSaker: () => new Promise((resolve, reject) => {}),
    fetchMeldinger: () => new Promise((resolve, reject) => {}),
    fetchSakstema: () => new Promise((resolve, reject) => {}),
    fetchOppfolgingNyKilde: () => new Promise((resolve, reject) => {}),
    fetchMeldekortNyKilde: () => new Promise((resolve, reject) => {}),
    fetchPersonNavnNyKilde: () => new Promise((resolve, reject) => {}),
    fetchPersonIdentNyKilde: () => new Promise((resolve, reject) => {}),
    fetchSakerNyKilde: () => new Promise((resolve, reject) => {}),
    fetchMeldingerNyKilde: () => new Promise((resolve, reject) => {}),
    fetchSakstemaNyKilde: () => new Promise((resolve, reject) => {}),
  };
};

const flushPromises = () => {
  return new Promise(resolve => setImmediate(resolve));
};

it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(wrapIntl(<App api={mockApi()}/>), div);
  ReactDOM.unmountComponentAtNode(div);
});

it('expect Login page rendering', () => {
  const api = mockApi();
  const component = ReactTestRenderer.create(wrapIntl(<App api={api} />));
  expect(component.toJSON()).toMatchSnapshot();
});

/* Temp while feature toggling <Hendelser> in <Infomeldinger>.
it('expect Oppfolging fetching', async () => {
  const api = mockApi();
  api.fetchOppfolging = () => new Promise((resolve, reject) => {
    resolve(
      {
        'erBrukerUnderOppfolging': false
      },
    );
  });
  const renderer = new ShallowRenderer();
  renderer.render(wrapIntl(<App api={api}/>));
  const component = renderer.getRenderOutput();
  await flushPromises();

  expect(component).toMatchSnapshot();
});

it('expect MeldekortInfo fetching', async () => {
  const api = mockApi();
  api.fetchMeldekort = () => new Promise((resolve, reject) => {
    resolve(
      {
        'etterregistrerteMeldekort': 0,
        'meldekortbruker': true,
        'nyeMeldekort': {
          'antallNyeMeldekort': 3,
          'nesteInnsendingAvMeldekort': null,
          'nesteMeldekort': {
            'fra': '2019-09-09',
            'kanSendesFra': '2019-09-21',
            'risikererTrekk': true,
            'sisteDatoForTrekk': '2019-09-30',
            'til': '2019-09-22',
            'uke': '37-38'
          }
        },
        'resterendeFeriedager': 0
      }
    );
  });
  const renderer = new ShallowRenderer();
  renderer.render(wrapIntl(<App api={api}/>));
  const component = renderer.getRenderOutput();
  await flushPromises();

  expect(component).toMatchSnapshot();
});

it('expect Personalia navn fetching', async () => {
  const api = mockApi();
  api.fetchPersonNavn = () => new Promise((resolve, reject) => {
    resolve(
      {
        'navn': 'VINAYAGUM-MASK AMIZIC'
      },
    );
  });
  const renderer = new ShallowRenderer();
  renderer.render(wrapIntl(<App api={api}/>));
  const component = renderer.getRenderOutput();
  await flushPromises();

  expect(component).toMatchSnapshot();
});

it('expect Personalia ident fetching', async () => {
  const api = mockApi();
  api.fetchPersonIdent = () => new Promise((resolve, reject) => {
    resolve(
      {
        'ident': 123
      },
    );
  });
  const renderer = new ShallowRenderer();
  renderer.render(wrapIntl(<App api={api}/>));
  const component = renderer.getRenderOutput();
  await flushPromises();

  expect(component).toMatchSnapshot();
});


it('expect PaabegynteSoknader fetching', async () => {
  const api = mockApi();

  api.fetchSaker = () => new Promise((resolve, reject) => {
    resolve(
      {
        'url': 'https://tjenester-t6.nav.no/',
        'antallPaabegynte': 2,
        'feilendeBaksystem': ['hello']
      }
    );
  });

  const renderer = new ShallowRenderer();
  renderer.render(wrapIntl(<App api={api}/>));
  const component = renderer.getRenderOutput();
  await flushPromises();

  expect(component).toMatchSnapshot();
});
 */
