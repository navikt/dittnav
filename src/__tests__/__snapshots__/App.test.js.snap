// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect Login page rendering 1`] = `
<main
  role="main"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="maincontent side-innhold"
      >
        <div
          className="col-md-12"
          id="dittnav-main-container"
        >
          <section
            className="infomeldinger-list"
          >
            <h1
              className="skjermleser"
            >
              <span>
                Varsler
              </span>
            </h1>
          </section>
          <div
            className="dittnav-lenkepanel-top-container"
          >
            <div
              className="dittnav-lenkepanel-liten"
              id={null}
            >
              <a
                alt="Dine saker"
                className="lenkepanel dittnav-lenkepanel-liten-item lenkepanel--border"
                href="http://localhost:9222/saksoversikt"
              >
                <span
                  className="typo-undertittel lenkepanel__heading"
                >
                  <span>
                    Dine saker
                  </span>
                </span>
                <span
                  className="lenkepanel__indikator"
                />
              </a>
              <a
                alt="Utbetalinger"
                className="lenkepanel dittnav-lenkepanel-liten-item lenkepanel--border"
                href="http://localhost:9222/utbetalingsoversikt"
              >
                <span
                  className="typo-undertittel lenkepanel__heading"
                >
                  <span>
                    Dine utbetalinger
                  </span>
                </span>
                <span
                  className="lenkepanel__indikator"
                />
              </a>
              <a
                alt="Innboks"
                className="lenkepanel dittnav-lenkepanel-liten-item last lenkepanel--border"
                href="http://localhost:9222/mininnboks"
              >
                <span
                  className="typo-undertittel lenkepanel__heading"
                >
                  <span>
                    Innboks
                  </span>
                </span>
                <span
                  className="lenkepanel__indikator"
                />
              </a>
            </div>
          </div>
          <h2
            className="typo-undertittel relatert-informasjon__subheader"
          >
            <span>
              Relatert informasjon
            </span>
          </h2>
          <div
            className="relatert-informasjon"
          >
            <nav
              className="relatert-informasjon__links"
            >
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="https://arbeidsplassen.nav.no/stillinger"
                >
                  Ledige stillinger
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="http://localhost:9222/pselv/publisering/uforetrygd.jsf?context=ut"
                >
                  Uføretrygd
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="https://foreldrepenger.nav.no"
                >
                  Dine foreldrepenger
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="http://localhost:9222/aktivitetsplan"
                >
                  Aktivitetsplan
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="http://localhost:9222/meldekort/om-meldekort"
                >
                  Meldekort
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="http://localhost:9222/veiledearbeidssoker/mistet-jobben/registrering"
                >
                  Registrer deg som arbeidssøker
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="https://stillingsok.nav.no/pam-stillingsok/lagrede-sok"
                >
                  Dine stillingssøk
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="http://localhost:9222/person/personopplysninger"
                >
                  Personopplysninger
                </a>
              </div>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;

exports[`expect MeldekortInfo fetching 1`] = `
<App
  api={
    Object {
      "fetchMeldekort": [Function],
      "fetchMeldinger": [Function],
      "fetchOppfolging": [Function],
      "fetchOppfolgingNyKilde": [Function],
      "fetchPersonIdent": [Function],
      "fetchPersonNavn": [Function],
      "fetchSaker": [Function],
      "fetchSakstema": [Function],
    }
  }
/>
`;

exports[`expect Oppfolging fetching 1`] = `
<App
  api={
    Object {
      "fetchMeldekort": [Function],
      "fetchMeldinger": [Function],
      "fetchOppfolging": [Function],
      "fetchOppfolgingNyKilde": [Function],
      "fetchPersonIdent": [Function],
      "fetchPersonNavn": [Function],
      "fetchSaker": [Function],
      "fetchSakstema": [Function],
    }
  }
/>
`;

exports[`expect PaabegynteSoknader fetching 1`] = `
<App
  api={
    Object {
      "fetchMeldekort": [Function],
      "fetchMeldinger": [Function],
      "fetchOppfolging": [Function],
      "fetchOppfolgingNyKilde": [Function],
      "fetchPersonIdent": [Function],
      "fetchPersonNavn": [Function],
      "fetchSaker": [Function],
      "fetchSakstema": [Function],
    }
  }
/>
`;

exports[`expect Personalia ident fetching 1`] = `
<App
  api={
    Object {
      "fetchMeldekort": [Function],
      "fetchMeldinger": [Function],
      "fetchOppfolging": [Function],
      "fetchOppfolgingNyKilde": [Function],
      "fetchPersonIdent": [Function],
      "fetchPersonNavn": [Function],
      "fetchSaker": [Function],
      "fetchSakstema": [Function],
    }
  }
/>
`;

exports[`expect Personalia navn fetching 1`] = `
<App
  api={
    Object {
      "fetchMeldekort": [Function],
      "fetchMeldinger": [Function],
      "fetchOppfolging": [Function],
      "fetchOppfolgingNyKilde": [Function],
      "fetchPersonIdent": [Function],
      "fetchPersonNavn": [Function],
      "fetchSaker": [Function],
      "fetchSakstema": [Function],
    }
  }
/>
`;
